### ISSUE TITLE
Incorrect "terraform FAILED" output when run against a directory which is not a terraform module

### ISSUE THREAD

**Issue by nk9** [tfverch/tfvc#34]
I installed version 0.7.2 through Homebrew. But this is what happens when I try to use it:

```
$ tfvc .
terraform FAILED Missing version constraints
─────────────────────────────────────────────────────

  Resolution
  ───────────────────────────────────────────────────
  Configure version constraints for terraform

  Details
  ───────────────────────────────────────────────────
  Type:                terraform
  Path:                .
  Name:                terraform
  Source:              github.com/hashicorp/terraform
  Version Constraints:
  Version:
  Latest Match:
  Latest Overall:      1.3.2

────────────────────────────────────────────────────
$ which terraform
/opt/homebrew/bin/terraform

$ terraform --version
Terraform v1.3.2
on darwin_arm64
```

If it matters, here's what `.` looks like:

```
.
├── README.md
├── environments
│   ├── common
│   │   ├── ecr.tf
│   │   ├── events.tf
│   │   ├── iam.tf
│   │   ├── lambda
│   │   │   └── resource-scheduler
│   │   │       ├── create_dist.sh
│   │   │       ├── dist.zip
│   │   │       └── src
│   │   │           ├── main.py
│   │   │           └── requirements.txt
│   │   ├── lambda.tf
│   │   ├── main.tf
│   │   ├── ses.tf
│   │   └── variables.tf
│   ├── prod
│   │   ├── main.tf
│   │   ├── outputs.tf
│   │   ├── project -> ../../modules/project
│   │   └── variables.tf
│   └── staging
│       ├── main.tf
│       ├── outputs.tf
│       ├── project -> ../../modules/project
│       └── variables.tf
├── justfile
├── modules
│   ├── project
│   │   ├── alb.tf
│   │   ├── cognito.tf
│   │   ├── context.tf
│   │   ├── database.tf
│   │   ├── ec2.tf
│   │   ├── ecs.tf
│   │   ├── iam.tf
│   │   ├── logs.tf
│   │   ├── main.tf
│   │   ├── network.tf
│   │   ├── outputs.tf
│   │   ├── redis.tf
│   │   ├── route_53.tf
│   │   ├── security_groups.tf
│   │   ├── ses.tf
│   │   └── variables.tf
│   └── vpc
│       ├── main.tf
│       ├── outputs.tf
│       └── variables.tf
└── utils
    └── vpc-dependencies.sh
```

**Comment by ryan-jan** [comment]
@nk9 `.` doesn't appear to be a terraform module looking at your directory tree. Given your project structure you'd need to run `tfvc` against one of your environment modules e.g. `tfvc ./environments/prod`.

You have however found a bug that I need to tweak in that  the main terraform version check should not be triggered in your situation given that there is not a valid terraform module in the specified directory. I'll sort this.

**Comment by nk9** [comment]
That makes sense! I didn't really take that from the error message, so hopefully your changes will add something a bit clearer in that situation. Thanks very much for the prompt reply.

When I do `tfvc environments/staging`, I am getting this warning instead of any output about the versions of the installed modules:

```
provider 'hashicorp/aws' WARNING Configured version does not match the latest available version
────────────────────────────────────────────────────────

  Resolution
  ──────────────────────────────────────────────────────────
  Consider using the latest version of this provider

  Details
  ────────────────────────────────────────────────────────
  Type:                provider
  Path:                environments/staging
  Name:                hashicorp/aws
  Source:              registry.terraform.io/hashicorp/aws
  Version Constraints: ~> 4.23.0
  Version:             4.23.0
  Latest Match:
  Latest Overall:      4.34.0

───────────────────────────────
```

Is that expected? If not, is it perhaps because all of my registry modules are declared in a module from a symlinked directory (see tree above)? I should point out that using symlinks wasn't actually my idea, IIRC it's the only way to reuse module code in a monorepo between multiple workspaces when using Terraform Cloud.

**Comment by ryan-jan** [comment]
@nk9 yes this output is expected. It shows that your staging environment is configured to use aws provider version `~> 4.23.0` (or in other words `4.23.x`) but there is a newer version of the provider available (`4.34.0`). Ideally you would configure your provider to use the latest version unless there was a reason you were pinning to `4.23.x`. For example you could use `~> 4.0` to allow `4.x.x` and run `terraform init -upgrade` to pull the latest version into you project.

Regarding you not seeing output for any modules, are you referencing any version controlled modules in your `staging` code, for example from the terraform registry or a remote git repo? `tfvc` does not provide any output for local modules as they are not version controlled and so there is no way of it determining if a local module is up to date. If you use modules from the registry or a git repo `tfvc` will try and determine if there is an update available. One other thing is that by default  `tfvc` only shows warnings and failures, to see everything that passed as well use the `-a` switch.

I'm working on some documentation which should hopefully make these things clearer from the offset for users it's just a work in progress at the moment.
